File: monorepo_python/packages/app/src/mono_app/cycle_b.py
# C1.b — other half of the cycle. Closes the loop back to Alpha.
from __future__ import annotations

from typing import TYPE_CHECKING
if TYPE_CHECKING:
    from .cycle_a import Alpha


class Bravo:
    def __init__(self, tag: str) -> None:
        self.tag = tag

    def bounce_to_alpha(self) -> str:
        from .cycle_a import Alpha                  # C1.b.runtime_import (deferred)
        return Alpha(f"bounce-from-{self.tag}").describe()

File: monorepo_typescript/apps/web/src/cycle_b.ts
// C1.b — closes the cycle back to Alpha.
import type { Alpha } from './cycle_a';

export class Bravo {
    constructor(public tag: string) {}

    async bounceToAlpha(): Promise<string> {
        const { Alpha } = await import('./cycle_a');
        return new Alpha(`bounce-from-${this.tag}`).describe();
    }
}
